django/contrib/auth/mixins.py

```diff
@@ -73,6 +74,15 @@ def dispatch(self, request, *args, **kwargs):
         return super().dispatch(request, *args, **kwargs)
 
 
+class LoginNotRequiredMixin:
```

Comment:
I might be missing something, but what's the reason for also having a mixin class? It seems like the `@login_not_required` decorator will work perfectly well for class-based views too no?

Reply:
I believe it largely comes down to personal preferences.